File: LeetCode-Solution/3940. Limit Occurrences in Sorted Array/3940.cpp
class Solution {
public:
    vector<int> limitOccurrences(vector<int>& nums, int k) {
       
        ios_base::sync_with_stdio(false);
        cin.tie(NULL);

        vector<int> ans;
        int n = nums.size();
        if (n == 0) return ans;

        int count = 1; 
        ans.push_back(nums[0]);

        for (int i = 1; i < n; i++) {
            if (nums[i] == nums[i - 1]) {
                count++; 
            } else {
                count = 1; 
            }
            if (count <= k) {
                ans.push_back(nums[i]);
            }
        }

        return ans;
    }
};

File: LeetCode-Solution/3940. Limit Occurrences in Sorted Array/3940.py
class Solution:
    def limitOccurrences(self, nums: list[int], k: int) -> list[int]:
        i = 0
        for n in nums:
            if i < k or nums[i - k] != n:
                nums[i] = n
                i += 1
        return nums[:i]